### PR TITLE
Deprecate setting index.optimize_auto_generated_id in 6.x

### PR DESCRIPTION
**Pull request by liketic**
Deprecate setting ```index.optimize_auto_generated_id``` in 6.x. 

Relates to #27600

### CODE REVIEW

**Comment by jasontedor** [comment]
test this please

**Comment by javanna** [comment]
@jasontedor the build succeeded when you ran it a (long) while ago, would you mind picking this one up?

**Review by jasontedor — CHANGES_REQUESTED** [review]

File: server/src/main/java/org/elasticsearch/index/engine/EngineConfig.java [hunk 1]
```diff
@@ -109,6 +114,7 @@
      * this setting won't be reflected re-enabled optimization until the engine is restarted or the index is closed and reopened.
      * The default is <code>true</code>
      */
+    @Deprecated
     public static final Setting<Boolean> INDEX_OPTIMIZE_AUTO_GENERATED_IDS = Setting.boolSetting("index.optimize_auto_generated_id", true,
         Property.IndexScope, Property.Dynamic);
```
[the remark this answers is not in the document]
Comment (liketic) [reply server/src/main/java/org/elasticsearch/index/engine/EngineConfig.java]: Thanks @jasontedor , please review again.

**Comment by jasontedor** [comment]
@liketic Would you update EngineConfigTests so that it compiles?

**Comment by jasontedor** [comment]
@elasticmachine test this please

**Review by jasontedor — APPROVED** [review]

**Comment by jasontedor** [comment]
@elasticmachine test this please

**Comment by jasontedor** [comment]
I added [v6.6.0](https://github.com/elastic/elasticsearch/labels/v6.6.0).

**Pull request merged**